tests/unit_tests/utils/lock_tests.py

```diff
@@ -32,56 +32,51 @@
 OTHER_KEY = get_key("ns2", a=1, b=2)
 
 
-def _get_lock(key: UUID, session: Session) -> Any:
+def _get_lock(key: UUID) -> Any:
     from superset.key_value.models import KeyValueEntry
 
-    entry = session.query(KeyValueEntry).filter_by(uuid=key).first()
+    entry = db.session.query(KeyValueEntry).filter_by(uuid=key).first()
     if entry is None or entry.is_expired():
         return None
 
     return JsonKeyValueCodec().decode(entry.value)
 
 
-def _get_other_session() -> Session:
```

Comment:
This isn't needed for these tests and causes issues when used with a nested transaction when we want to rollback to a previous `SAVEPOINT`.

Reply:
We do need to ensure that the lock is committed to the metastore somehow. But let me do the key value DAO refactor first, that might help clean up this test. In the interim, feel free to relax/disable the test if needed.